metricbeat/module/jolokia/jmx/data.go

```diff
@@ -136,11 +135,13 @@ func eventMapping(entries []Entry, mapping AttributeMapping) ([]mapstr.M, error)
 					errs = append(errs, err)
 				}
 			case map[string]interface{}:
-				constructEvents(entryValues, v, mbeanEvents, mapping, errs)
+				errs = constructEvents(entryValues, v, mbeanEvents, mapping, errs)
 			}
 		case []interface{}:
-			entryValues := v.Value.(map[string]interface{})
-			constructEvents(entryValues, v, mbeanEvents, mapping, errs)
+			entryValues, ok := v.Value.(map[string]interface{})
+			if ok {
```

Comment:
Not my code to say so either way, but I wonder if this needs an `else` to add an error?